integrationtests/testground/main.go

```diff
@@ -198,39 +201,40 @@ func testRound(env *runtime.RunEnv, ic *run.InitContext, round string, desiredAd
 		}
 		msg("Joined thread")
 
+		ic.SyncClient.MustSignalAndWait(ctx, sync.State("ready-"+round+"-phase1"), livePeers)
 		start := time.Now()
-		doneStep1 := sync.State("done-" + round + "-step1")
+		donePhase1 := sync.State("done-" + round + "-phase1")
 		go func() {
-			receivedRecords = thr.WaitForRecords(ctx, expectedRecords)
-			env.R().RecordPoint("round-"+round+"-step-1-elapsed-seconds", time.Since(start).Seconds())
-			msg("Peer #%d done %s network step 1", ic.GlobalSeq, round)
-			ic.SyncClient.MustSignalAndWait(ctx, doneStep1, livePeers)
+			_ = thr.WaitForRecords(ctx, recordsToReceive)
+			env.R().RecordPoint("round-"+round+"-phase-1-elapsed-seconds", time.Since(start).Seconds())
+			msg("Peer #%d done %s network phase 1, received %d records", ic.GlobalSeq, round, recordsToReceive)
+			ic.SyncClient.MustSignalAndWait(ctx, donePhase1, livePeers)
 		}()
 
-		for i, prev := 0, cid.Undef; i < myRecords; i++ {
-			rec, err := thr.AddRecord(ctx, prev)
-			if err != nil {
-				return err
-			}
-			prev = rec.Cid()
+		if err := thr.CreateRecords(ctx, recordsToSend); err != nil {
```

Comment:
moved the loop into `CreateRecords`, as it doesn't (incorrectly) manipulate `Prev` anymore.

Reply:
👍

Reply:
We could also use the raw `CreateRecord` API: https://github.com/textileio/go-threads/blob/master/core/net/net.go#L60 Or maybe that's a different/future test.

Reply:
Oh right. will do

Reply:
Done https://github.com/textileio/go-threads/pull/523/commits/4c9ab518efccaccbdae274bb280f2ae7a9ba5cf8